index.js

```diff
@@ -1,53 +1,54 @@
-const debug = require('debug')('x-ray:nightmare');
-const normalize = require('normalizeurl');
-const Nightmare = require('nightmare');
+const debug = require('debug')('x-ray:puppeteer');
+// const normalize = require('normalizeurl');
+const puppeteer = require('puppeteer');
 
 
 function makeDriver(opts) {
   opts = opts || {};
   // add default user agent
-  if (!opts.useragent) opts.useragent = 'xray/nightmare';
+  if (!opts.useragent) opts.useragent = 'xray/puppeteer';
 
-	return (ctx, callback) => {
-		debug('going to %s', ctx.url);
-
-    const nightmare = new Nightmare(opts);
-    this.instance = nightmare;
+  return (ctx, callback) => {
+    debug('going to %s', ctx.url);
+    this.instance = puppeteer;
 
     // Setup standard listeners
-    nightmare
-      .useragent(opts.useragent)
-      .on('error', (msg) => debug('client-side javascript error %s', msg))
-      .on('timeout', (timeout) => callback(new Error(timeout)))
-      .on('resourceReceived', (resource) => {
-        if (normalize(resource.url) == normalize(ctx.url)) {
-          debug('got response from %s: %s', resource.url, resource.status);
-          ctx.status = resource.status;
-        }
-      })
-      .on('urlChanged', (url) => {
-        debug('redirect: %s', url);
-        ctx.url = url;
-      });
-    
-    // Basic operation is to get the rendered page content
-    // and return it as the "complete" document.
-    //
-    // X-ray will then use Cheerio on this document, so the same queries & filters
-    // that work with request-based driver will work here (ie. NOT nightmare specific).
-    nightmare
-      .goto(ctx.url)
-      .evaluate(() => {
-        // Get the entire document contents
-        return document.documentElement.outerHTML;
+    // nightmare
+    //   .useragent(opts.useragent)
+    //   .on('error', (msg) => debug('client-side javascript error %s', msg))
+    //   .on('timeout', (timeout) => callback(new Error(timeout)))
+    //   .on('resourceReceived', (resource) => {
+    //     if (normalize(resource.url) == normalize(ctx.url)) {
+    //       debug('got response from %s: %s', resource.url, resource.status);
+    //       ctx.status = resource.status;
+    //     }
+    //   })
+    //   .on('urlChanged', (url) => {
+    //     debug('redirect: %s', url);
+    //     ctx.url = url;
+    //   });
```

Comment:
Still needs to be clean and left this here to know what I should add like `normalizeurl`